app/components/SearchableMetricSelector.js

```diff
@@ -58,11 +92,109 @@ function Modal(props){
     )
 }
 
+function SimpleMetricSearchResults(props){
+  const { searchText, supportedMetrics, onClose, selectMetric } = props
+  const searchTextLower = searchText.toLowerCase().trim()
+  const searchResults = searchTextLower.length === 0 
+    ? Object.keys(supportedMetrics)
+    : Object.keys(supportedMetrics)
+      .filter((k) => {
+        const metric = supportedMetrics[k]
+        const metricNameLower = metric.name.toLowerCase()
+        return metricNameLower.indexOf(searchTextLower) > -1
+      })
+      
+  return ( 
+    <div className="searchResults">
+      {searchResults.map((k, i) => (
+        <p
+          className="metricResult"
+          key={i} 
+          onClick={ () => {
+            selectMetric(k)
+            onClose()  
+          }}
+        >
+          {supportedMetrics[k].name}
+            <ul className="metricMetaData">
+              <li>Dataset: {supportedMetrics[k].dataset}</li>
+              <li>Description: {supportedMetrics[k].description}</li>
+            </ul>
+          </p>
+              ))}
+    </div>
+  )
+  
+}
+
+function GroupedMetricSearchResults(props){
+  const { searchText, supportedMetrics, onClose, selectMetric, tagDictionary, tagDictArray } = props
+  const searchTextLower = searchText.toLowerCase().trim()
+  const searchResults = searchTextLower.length === 0 
+    ? Object.keys(supportedMetrics)
+    : Object.keys(supportedMetrics)
+      .filter((k) => {
+        const metric = supportedMetrics[k]
+        const metricNameLower = metric.name.toLowerCase()
+        return metricNameLower.indexOf(searchTextLower) > -1
```

Comment:
Are you also going to consider the dataset name in the search filter? It's okay if not, that can be done later or not done at all.

We may also want to include metric description in the search.

Either way, it's possible that for both `SimpleMetricSearchResults` and `GroupedMetricSearchResults`, the search filter function has exactly the same logic. In that case, you can reorganize it into a common helper function and then call that function in each component like this:

```js
const searchResults = filterMetricsForSearch(supportedMetrics, searchText);
```